salt/_logging/impl.py

```diff
@@ -288,37 +288,31 @@ def _log(
         else:
             extra["exc_info_on_loglevel"] = exc_info_on_loglevel
 
-        try:
-            logging._acquireLock()
```

Comment:
Do you know why this was here in the first place? Are you certain that we don't add a race condition by removing the lock?

Reply:
I tested this change with stress test running for slightly more than 24 hours with no any issues, if I keep this lock acquire and release in place it leads to deadlocks on calling `logging.getLogger(__name__)` in `salt-api` child processes almost randomly in few minutes up to few hours, but finally `salt-api` stops responding on incoming requests and keeps connections open from taskomatic ssh push.
The worst thing that deadlocks on logging and importlib modules with using multiprocessing are known as a bugs for a long time, but still no any straight solution, just the recommendation for logging to process all the log messages with distinct process passing messages for logging through the `multiprocessing.Queue`.

Reply:
I guess without the locks the logs will be out of order or what's the original purpose behind them?
I just want to make sure we understand what we're changing. Could you share how you ran the stress test and its scope?

Reply:
It runs a lot of simulataneous events with state.apply on salt-ssh systems. The state is the following:
```
dummy-test:
  cmd.run:
  - name: for (( i=1; i<=5; i++ )); do  printf "DUMMYOUTPUT"; sleep 1; done
```
The state apply should take some time and multiple simultaneous ssh sessions causing the issues for the customer. I tried to reproduce the issues in amplified syntetic form.